src/compiler/emitter.ts

```diff
@@ -4397,6 +4396,21 @@ namespace ts {
             return lines;
         }
 
+        function writeLineSeparatorsAndIndentBefore(node: Node, parent: Node): boolean {

```

Comment:
These functions are extracted from `emitParenthesizedExpression` because fixing the actual bug revealed that a JSX example was only working by accident because of the bug, and needed to borrow this code from `emitParenthesizedExpression`.